integration/integration_test.go

```diff
@@ -6700,3 +6704,100 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
 	require.NoError(t, err)
 	require.Equal(t, testFilePath, sftpEvent.GetString(events.SFTPPath))
 }
+
+// TestProxySSHPortMultiplexing ensures that the Proxy SSH port
+// is serving both SSH and gRPC regardless of TLS Routing mode.
+func TestProxySSHPortMultiplexing(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name              string
+		disableTLSRouting bool
+	}{
+		{
+			name: "TLS routing enabled",
+		},
+		{
+			name:              "TLS routing disabled",
+			disableTLSRouting: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			privateKey, publicKey, err := testauthority.New().GenerateKeyPair()
+			require.NoError(t, err)
+
+			rc := helpers.NewInstance(t, helpers.InstanceConfig{
+				ClusterName: "example.com",
+				HostID:      uuid.New().String(),
+				NodeName:    Host,
+				Priv:        privateKey,
+				Pub:         publicKey,
+				Log:         utils.NewLoggerForTests(),
+			})
+
+			rcConf := service.MakeDefaultConfig()
+			rcConf.DataDir = t.TempDir()
+			rcConf.Auth.Preference.SetSecondFactor("off")
+			rcConf.SSH.Enabled = false
+			rcConf.Proxy.DisableWebInterface = true
+			rcConf.Proxy.DisableTLS = false
+			rcConf.Proxy.DisableALPNSNIListener = test.disableTLSRouting
+			rcConf.CircuitBreakerConfig = breaker.NoopBreakerConfig()
+
+			// Create identical user/role in both clusters.
+			me, err := user.Current()
+			require.NoError(t, err)
+
+			role := services.NewImplicitRole()
+			role.SetName("test")
+			role.SetLogins(types.Allow, []string{me.Username})
+			role.SetNodeLabels(types.Allow, map[string]apiutils.Strings{types.Wildcard: []string{types.Wildcard}})
+			rc.AddUserWithRole(me.Username, role)
+
+			err = rc.CreateEx(t, nil, rcConf)
+			require.NoError(t, err)
+
+			require.NoError(t, rc.Start())
+			t.Cleanup(func() {
+				require.NoError(t, rc.StopAll())
+			})
+
+			// create an authenticated client for the user
+			tc, err := rc.NewClient(helpers.ClientConfig{
+				Login:   me.Username,
+				Cluster: helpers.Site,
+				Host:    Host,
+			})
+			require.NoError(t, err)
+
+			// connect via SSH
+			ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
```

Comment:
```suggestion
			ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
			defer cancel()
```

so it doesn't hang around on errors.